pkg/server/tls/authority/authority.go

```diff
@@ -169,6 +185,10 @@ func (d *DynamicAuthority) Sign(template *x509.Certificate) (*x509.Certificate,
 	d.signMutex.Lock()
 	defer d.signMutex.Unlock()
 
+	if d.currentCertData == nil || d.currentPrivateKeyData == nil {
+		return nil, fmt.Errorf("no tls.Certificate available yet, try again later")
```

Comment:
I added a separate error message for this situation to make the logged messages more insightful.

Reply:
Nice! Possibly related issue https://github.com/cert-manager/cert-manager/issues/7138.